plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -86,7 +85,8 @@ private CloudSdk getCloudSdk() {
     MessageConsole messageConsole =
         MessageConsoleUtilities.getMessageConsole(CONSOLE_NAME, null, true /* show */);
     final MessageConsoleStream outputStream = messageConsole.newMessageStream();
-    CloudSdk cloudSdk = new CloudSdkProvider().createBuilder()
+    CloudSdk cloudSdk =
```

Comment:
I also think we should be very careful having Jobs that trigger UI actions.  This could open the _Console_ view.

This code also doesn't handle the case where there's no Cloud SDK to be found.  Imagine a user in my situation where the Cloud SDK is not located in one of the spots checked by `appengine-plugin-core`'s `*Resolver`s.  I've import an existing project, and click _Deploy_.  This code will show an `AppEngineException` as there was no `sdkPath` specified and the SDK couldn't be found.  So you could change it to use `CloudSdkPrompter`, but now your job can be held up by UI interruption. 

I'm just advocating moving this particular method into the handler.  Obtain the and configure `CloudSdk` instance in the deploy handler via `CloudSdkPrompter` with the console listeners, and then provide the `CloudSdk` instance to the job.  When it comes down to it, the logging of the console output is orthogonal to the deploy operation.